src/util/simplify_expr2.cpp

```diff
@@ -415,13 +414,87 @@ static type2tc common_arith_op2_type(expr2tc &e, expr2tc &f)
 
 expr2tc add2t::do_simplify() const
 {
+  // (A + 1) + ~B --> A - B
+  if(is_add2t(side_1) && is_bitnot2t(side_2))
+  {
+    dump();
+    auto sidecheck = to_add2t(side_1).side_2;
+    auto sidecheck2 = to_add2t(side_1).side_1;
+    if(is_constant_int2t(sidecheck) == 1)
```

Comment:
is_constant_int returns a boolean value.

If you want to check if the value itself is 1, then use `to_constant_int(sidecheck).value`

Reply:
i just changed this on the lastest version by change the line  430 to if(to_constant_int2t(sidecheck).value == 1)